qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteSnippetCompletionInWhenSectionTest.java

```diff
@@ -54,19 +55,45 @@ public void isSnippetCompletionInNestedWhen() throws Exception {
 				"{/when}";
 		testCompletionFor(template, //
 				true, // snippet support
-				SECTION_SNIPPET_SIZE /* #each, #if, ... */ + 1, /* #is */ //
+				SECTION_SNIPPET_SIZE /* #each, #if, ... */ + 2, /* #is, #else */ //
 				c("is", "{#is ${1:case}}$0", r(1, 1, 1, 5)));
 	}
 
 	@Test
-	public void isSnippetCompletionInNestedWhenWithIs() throws Exception {
+	public void snippetCompletionInNestedWhenWithIs() throws Exception {
 		String template = "{#when value}\n" + //
 				"\t{#is case}\n" + //
 				"\t|\n" + //
 				"{/when}";
 		testCompletionFor(template, //
 				true, // snippet support
-				SECTION_SNIPPET_SIZE /* #each, #if, ... */ + 1, /* #is */ //
-				c("is", "{#is ${1:case}}$0", r(2, 1, 2, 1)));
+				SECTION_SNIPPET_SIZE /* #each, #if, ... */ + 2, /* #is, #else */ //
+				c("is", "{#is ${1:case}}$0", r(2, 1, 2, 1)), //
+				c("else", "{#else}$0", r(2, 1, 2, 1)));
+	}
+
+	@Test
+	public void elseSnippetCompletionInNestedWhenWithIs() throws Exception {
+		String template = "{#when value}\n" + //
+				"\t{#is case}\n" + //
+				"\t{#el|}\n" + //
+				"{/when}";
+		testCompletionFor(template, //
+				true, // snippet support
+				SECTION_SNIPPET_SIZE /* #each, #if, ... */ + 2, /* #is, #else */ //
+				c("else", "{#else}$0", r(2, 1, 2, 6)));

```

Comment:
Add is here too

Reply:
Since the completion is for "\t{#el|}\n", should there only be the "else" completion?

Reply:
No, because we don't manage filter on server side. Teh filter is done in client side which manages fuzzy filter which is very nice.